parser.py

```diff
@@ -422,18 +434,23 @@ def course_attrs(self):
                 if box_title == COURSE_DETAIL:
                     # Units and course components
                     labels = table.find_all("span", {"class": EDITBOX_LABEL_CLASS})
+                    
                     data = table.find_all("span", {"class": EDITBOX_DATA_CLASS})
+
+                    dx = 0
                     for x in range(0, len(labels)):
                         if labels[x].string == COURSE_COMPS:
                             # Last datafield, has multiple type -> value mappings
                             comp_map = {}
-                            for i in range(x, len(data), 2):
+                            for i in range(x, x+(len(data)-len(labels)), 2):
                                 comp_map[data[i].string] = data[i+1].get_text()
+                                dx = i+2
```

Comment:
Why are you setting this every iteration of the loop? Won't it just have a known value, as AFAICT it isn't used in the loop?